yt/yt/library/profiling/solomon/cube.h

```diff
@@ -23,18 +24,15 @@ struct TReadOptions
 
     std::function<bool(const std::string&)> SensorFilter;
 
-    bool ConvertCountersToRateGauge = false;
-    bool ConvertCountersToDeltaGauge = false;
-    bool RenameConvertedCounters = true;
-    double RateDenominator = 1.0;
-    bool EnableHistogramCompat = false;
-
-    bool EnableSolomonAggregationWorkaround = false;
+    // NB: Be careful when modifying this field, as you might be modifying
+    // a shared object. If you need to modify it, make a deep copy first.
+    // TODO(achulkov2): Ideally, it should be a lite YSON struct, but I do
+    // not know how to use such structs as bases for ref-counted YSON structs.
+    TScrapeOptionsPtr ScrapeOptions;
```

Comment:
Naming-wise, I would prefer not to use the term "scrape" here, even though it is well-known in the industry. The reason is we already have a [probably not the best] term for this operation: "read sensors". So it feels natural to name this new struct `TReadOptions`.

The problem is this name is already taken by the outer struct. Wrongfully taken, I believe, as essentially this outer struct is a bunch of parameters which we've consolidated to not have methods with millions of arguments. We call these structs "contexts" in our codebase, so I suggest the following renaming: `TReadOptions -> TReadSensorsContext`, `TScrapeOptions -> TReadOptions`.

Reply:
Nice names, I think I can live with them just fine!